packages/builders/src/messages/embed/UnsafeEmbed.ts

```diff
@@ -138,7 +192,8 @@ export class UnsafeEmbed implements APIEmbed, JSONEncodable<APIEmbed> {
 	 * @param fields The replacing field objects
 	 */
 	public spliceFields(index: number, deleteCount: number, ...fields: APIEmbedField[]): this {
-		this.fields.splice(index, deleteCount, ...UnsafeEmbed.normalizeFields(...fields));
+		this.data.fields ??= [];
+		this.data.fields.splice(index, deleteCount, ...UnsafeEmbed.normalizeFields(...fields));
```

Comment:
More optimization! `Array#splice` sets all the elements passed via rest when the source array is empty, so we can optimize this by just making the fields array be the passed fields.
```suggestion
		fields = UnsafeEmbed.normalizeFields(...fields);

		if (this.data.fields) this.data.fields.splice(index, deleteCount, ...fields);
		else this.data.fields = fields;
```